department-of-veterans-affairs/caseflow
# Craig/appeals 13264

Resolves [APPEALS-13264](https://vajira.max.gov/browse/APPEALS-13264)

### Description
- Added save route and methods to routes.rb and cavc_dashboard_controller
- Added save functionality to front end (actions, reducer)
- Updated cavc_dashboard_disposition and cavc_dashboard_issue to cascade destroy
- Changed AddCavcDashboardIssue modal to create a formatted issue and dashboard_disposition as if they came from the back end
- Added 'initial_state' to redux store for conditionally enabling save button
- Added unit and controller tests

### Acceptance Criteria
- [ ] Adding and removing issues on the CAVC Dashboard can be saved

### Testing Plan
1. Go to any CAVC Dashboard as an OAI user
2. Add a CAVC Dashboard issue
3. Click 'Save Changes'
4. Verify that the user is redirected back to the case details page
5. Go back into the CAVC Dashboard
6. Verify that the added issue loads and displays with disposition value
7. Remove the added issue
8. Click 'Save Changes'
9. Go back into the CAVC Dashboard
10. Verify that the removed issue does not load
